core/src/components/input-otp/input-otp.tsx

```diff
@@ -0,0 +1,451 @@
+import type { ComponentInterface, EventEmitter } from '@stencil/core';
+import { Component, Element, Event, Host, Prop, State, h, Watch } from '@stencil/core';
+import { createColorClasses } from '@utils/theme';
+
+import { getIonMode } from '../../global/ionic-global';
+import type { Color } from '../../interface';
+
+import type { InputOtpChangeEventDetail, InputOtpCompleteEventDetail } from './input-otp-interface';
+
+@Component({
+  tag: 'ion-input-otp',
+  styleUrls: {
+    ios: 'input-otp.ios.scss',
+    md: 'input-otp.md.scss',
+  },
+  scoped: true,
```

Comment:
What is the reason for making this a scoped component vs a shadow one?

Reply:
I chose to make it a `scoped` component to match [input](https://github.com/ionic-team/ionic-framework/blob/7d639b0412120523f758942c855cb69f9a52e9d9/core/src/components/input/input.tsx#L44). In our [latest research](https://github.com/ionic-team/ionic-framework-design-documents/blob/main/projects/ionic-framework/components/input/0005-shadow-dom.md) we found that we couldn't convert input to shadow because there are a couple of properties that don't work with shadow. One of them being `autocomplete` which is used [here](https://github.com/ionic-team/ionic-framework/blob/97649a006b35a7cde236857473961d96c503fa8b/core/src/components/input-otp/input-otp.tsx#L473). I don't think we are able to move components using native `input` elements to shadow yet.